### ISSUE TITLE
Sub Editor does not saves updated autocalc price when saving sub changes

### ISSUE THREAD

**Issue by Daimonin** [Regalis11/Barotrauma#4127]
Game Version: 0.10.5.1

When making edits to an existing submarine, the editor calculates the price automatically, and presents it in the save dialog. It does not however actually save this price unless the user interacts with the price field. 

This leads to subs being sold in single player and multiplayer campaigns at the cost of the original sub before edits were made, or at a price of 0 if the user originally saved the sub before actually placing any objects.

In order to fix the sub price, the user has to open the sub in editor, change the auto calculated price, save, then change the price back to auto calculated value and save again. It is possible that just clicking on the field and pressing enter may work, I did not think of that during testing. The user must keep doing this any time they make a change to the sub that would increase it's auto-calculated cost.

**In short:** The save function in sub editor appears to only save the sub price when the user changes the price, not when the editor auto-calculation changes the price.

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/b54383d8e75cc24af80f2e398eb2d6057fd3cc14

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.
